pkg/netpol/connlist/connlist.go

```diff
@@ -384,21 +416,60 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine) ([]Peer2Pe
 			}
 			allowedConnections, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer)
 			if err != nil {
-				ca.errors = append(ca.errors, newResourceEvaluationError(err))
 				return nil, err
 			}
 			// skip empty connections
 			if allowedConnections.IsEmpty() {
 				continue
 			}
-			connectionObj := &connection{
+			p2pConnection := &connection{
 				src:               srcPeer,
 				dst:               dstPeer,
 				allConnections:    allowedConnections.AllConnections(),
 				protocolsAndPorts: allowedConnections.ProtocolsAndPortsMap(),
 			}
-			res = append(res, connectionObj)
+			res = append(res, p2pConnection)
+		}
+	}
+	return res, nil
+}
+
+// getIngressAllowedConnections returns connections list from IngressAnalyzer intersected with PolicyEngine's connections
+func (ca *ConnlistAnalyzer) getIngressAllowedConnections(ia *ingressanalyzer.IngressAnalyzer,
+	pe *eval.PolicyEngine, peerList []eval.Peer) ([]Peer2PeerConnection, error) {
+	res := make([]Peer2PeerConnection, 0)
+	ingressConns, err := ia.AllowedIngressConnections()
+	if err != nil {
+		return nil, err
+	}
+	// adding the ingress controller pod to the policy engine,
+	ingressControllerPod, err := pe.AddPodByNameAndNamespace(ingressanalyzer.IngressPodName, ingressanalyzer.IngressPodNamespace)
+	if err != nil {
+		return nil, err
+	}
+	for _, peer := range peerList {
+		// if there is no ingress connections to the peer - skip
+		if _, ok := ingressConns[peer.String()]; !ok {
+			continue
+		}
+		// otherwise, compute allowed connections based on pe.policies to the peer, then intersect the conns with
+		// ingress connections to the peer -> the intersection will be appended to the result
+		peConn, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(ingressControllerPod, peer)
+		if err != nil {
+			return nil, err
+		}
+		ingressConns[peer.String()].(*common.ConnectionSet).Intersection(peConn.(*common.ConnectionSet))
+		if ingressConns[peer.String()].IsEmpty() {
+			ca.logger.Warnf("Ingrees Connections to: " + peer.String() + " were blocked by network-policies")
```

Comment:
change the warning string: "Analysis of Ingress/Route resources inferred there is a configured connectivity from ingress-controller to workload [name], but network policies may be blocking access to this workload (unless specifically permitted to the installed ingress controller pod)"